vlm/src/main/scala/geotrellis/contrib/vlm/RasterSource.scala

```diff
@@ -77,37 +76,18 @@ trait RasterSource extends CellGrid[Long] with Serializable {
   /** Raster pixel row count */
   def rows: Long = gridExtent.rows
 
-  /** Reproject to different CRS with explicit sampling reprojectOptions.
-    * @see [[geotrellis.raster.reproject.Reproject]]
-    * @group reproject
-    */
-  def reproject(crs: CRS, reprojectOptions: Reproject.Options, strategy: OverviewStrategy): RasterSource
-
-  /** Sampling grid is defined over the footprint of the data at default resolution
-    *
-    * When using this method the cell size in target CRS will be estimated such that
-    * each cell in target CRS has ''approximately'' the same geographic coverage as a cell in the source CRS.
-    *
-    * @group reproject
-    */
-  def reproject(crs: CRS, method: ResampleMethod = NearestNeighbor, strategy: OverviewStrategy = AutoHigherResolution): RasterSource =
-    reproject(crs, Reproject.Options(method = method), strategy)
+  def reproject(targetCRS: CRS, resampleGrid: Option[ResampleGrid[Long]] = None, method: ResampleMethod = NearestNeighbor, strategy: OverviewStrategy = AutoHigherResolution): RasterSource
```

Comment:
Its very strange to see `Option` of `ResampleGrid[Long]`, `ResampleGrid` being a little ADT of possible ways to resample there could be a `DefaultResample` or `AutoResample` that fits the same purpose. Possible the name needs to change to give that meaning more clarity, but definitely it can encode the idea of default behavior.

Reply:
Damn, I don't have an explantation why haven't I done this.